File: leetcode2/1easy/유시연/2273.py
def remove_anagrams(words):
    changed = True
    while changed:
        changed = False
        for i in range(len(words) - 1, 0, -1): 
            if len(words[i]) == len(words[i - 1]) and sorted(words[i]) == sorted(words[i - 1]):
                del words[i] 
                changed = True
                break  
    return words

File: leetcode/1easy/박은비/2273.swift
class Leetcode.이재훈.이재훈.Solution {
    @_optimize(speed)
    func removeAnagrams(_ words: [String]) -> [String] {

        var answer = words
        var i = 1
// 한 작업에서 0 < i < words.length이고
// words[i - 1]과 words[i]가 애너그램인 모든 인덱스 i를 선택하고 words에서 words[i]를 삭제합니다. 조건을 만족하는 인덱스를 선택할 수 있는 한 이 작업을 계속 수행합니다.
        while i < answer.count {

            if answer[i-1].sorted() == answer[i].sorted() {
                answer.remove(at: i)
            } else {
                i += 1
            }
        }
        
        return answer
    }
}
